easybuild/tools/filetools.py

```diff
@@ -503,62 +504,42 @@ def get_local_dirs_purged():
 
 def extract_cmd(filepath, overwrite=False):
     """
-    Determines the file type of file fn, returns extract cmd
-    - based on file suffix
-    - better to use Python magic?
+    Determines the file type of file at filepath, returns extract cmd based on file suffix
     """
     filename = os.path.basename(filepath)
-    exts = [x.lower() for x in filename.split('.')]
-    target = '.'.join(exts[:-1])
-    cmd_tmpl = None
-
-    # gzipped or gzipped tarball
-    if exts[-1] in ['gz']:
-        if exts[-2] in ['tar']:
-            # unzip .tar.gz in one go
-            cmd_tmpl = "tar xzf %(filepath)s"
-        else:
-            cmd_tmpl = "gunzip -c %(filepath)s > %(target)s"
-
-    elif exts[-1] in ['tgz', 'gtgz']:
-        cmd_tmpl = "tar xzf %(filepath)s"
-
-    # bzipped or bzipped tarball
-    elif exts[-1] in ['bz2']:
-        if exts[-2] in ['tar']:
-            cmd_tmpl = 'tar xjf %(filepath)s'
-        else:
-            cmd_tmpl = "bunzip2 %(filepath)s"
-
-    elif exts[-1] in ['tbz', 'tbz2', 'tb2']:
-        cmd_tmpl = "tar xjf %(filepath)s"
-
-    # xzipped or xzipped tarball
-    elif exts[-1] in ['xz']:
-        if exts[-2] in ['tar']:
-            cmd_tmpl = "unxz %(filepath)s --stdout | tar x"
-        else:
-            cmd_tmpl = "unxz %(filepath)s"
-
-    elif exts[-1] in ['txz']:
-        cmd_tmpl = "unxz %(filepath)s --stdout | tar x"
-
-    # tarball
-    elif exts[-1] in ['tar']:
-        cmd_tmpl = "tar xf %(filepath)s"
-
-    # zip file
-    elif exts[-1] in ['zip']:
-        if overwrite:
-            cmd_tmpl = "unzip -qq -o %(filepath)s"
-        else:
-            cmd_tmpl = "unzip -qq %(filepath)s"
-
-    elif exts[-1] in ['iso']:
-        cmd_tmpl = "7z x %(filepath)s"
+    pat = r'\.(?P<ext>tar\.gz|gz|tgz|gtgz|tar.bz2|bz2|tbz|tbz2|tb2|tar.xz|xz|txz|tar|zip|iso)$'
+    try:
+        ext = '.' + re.search(pat, filename, flags=re.IGNORECASE).group('ext')
+    except AttributeError:
+        raise EasyBuildError('Unknown file type for file %s (%s)', filepath, ext)
+
+    target = filename.rstrip(ext)
+
+    extract_cmds = {
+        # gzipped or gzipped tarball
+        '.tar.gz':  "tar xzf %(filepath)s",
+        '.gz':      "gunzip -c %(filepath)s > %(target)s",
+        '.tgz':     "tar xzf %(filepath)s",
+        '.gtgz':    "tar xzf %(filepath)s",
+        # bzipped or bzipped tarball
+        '.tar.bz2': "tar xjf %(filepath)s",
+        '.bz2':     "bunzip2 %(filepath)s",
+        '.tbz':     "tar xjf %(filepath)s",
+        '.tbz2':    "tar xjf %(filepath)s",
+        '.tb2':     "tar xjf %(filepath)s",
+        # xzipped or xzipped tarball
+        '.tar.xz':  "unxz %(filepath)s --stdout | tar x",
+        '.xz':      "unxz %(filepath)s",
+        '.txz':     "unxz %(filepath)s --stdout | tar x",
```

Comment:
@sfranky: are you up for fixing the remarks yourself (just push extra commits to your branch, this PR will be updated automagically)

if not, let us know, so we can handle it and get this merged